lib/test.js

```diff
@@ -93,7 +93,22 @@ Test.prototype.run = function () {
         this.timeoutAfter(this._timeout);
     }
     this.emit('prerun');
-    this._cb(this);
+    var callbackReturn = this._cb(this);
+
+    if (
+        typeof Promise === 'function' &&
+        callbackReturn &&
+        typeof callbackReturn.then === 'function' &&
+        typeof callbackReturn.catch === 'function'
+    ) {
+        callbackReturn.catch(function onError(err) {
```

Comment:
since it might not be a real promise, we should wrap it in Promise.resolve first.

Reply:
I don't understand why that matters in this concrete case. It might be a good practice in general but I think for this patch it's unnecessary.